eng/mgmt/automation/generate_data.py

```diff
@@ -56,7 +111,16 @@ def generate(
     return True
 
 
-def compile_package(output_dir):
+def install_build_tools(sdk_root: str):
+    command = 'mvn --no-transfer-progress clean install -f {0} -pl com.azure:sdk-build-tools'.format(os.path.join(sdk_root, 'pom.xml'))
+    logging.info(command)
+    if os.system(command) != 0:
+        logging.error('[COMPILE] Maven build fail for sdk-build-tools')
+        return False
+    return True
+
+
+def compile_package(output_dir: str):
     command = 'mvn --no-transfer-progress clean verify package -f {0}'.format(os.path.join(output_dir, 'pom.xml'))
```

Comment:
Currently the script does not add all necessary configure for new module (version_client.txt, root POM, etc.), hence we only build within the module, not from the root POM.